sonar-python-plugin/src/main/java/org/sonar/plugins/python/indexer/SonarQubePythonIndexer.java

```diff
@@ -54,21 +56,21 @@ public class SonarQubePythonIndexer extends PythonIndexer {
   private final List<InputFile> testFiles = new ArrayList<>();
   private final Map<InputFile, String> inputFileToFQN = new HashMap<>();
 
-  public SonarQubePythonIndexer(List<InputFile> inputFiles, CacheContext cacheContext) {
+  public SonarQubePythonIndexer(List<InputFile> inputFiles, CacheContext cacheContext, SensorContext context) {
+    this.projectBaseDirAbsolutePath = context.fileSystem().baseDir().getAbsolutePath();
+    this.caching = new Caching(cacheContext, getCacheVersion(context));
     inputFiles.forEach(f -> {
       if (f.type().equals(InputFile.Type.MAIN)) {
         mainFiles.add(f);
       } else {
         testFiles.add(f);
       }
     });
```

Comment:
I would suggest to avoid the foreach at line 69 and move the `put` operation while iterating over `inputFiles`
```suggestion
    inputFiles.forEach(f -> {
      if (f.type().equals(InputFile.Type.MAIN)) {
        mainFiles.add(f);
        inputFileToFQN.put(f, SymbolUtils.fullyQualifiedModuleName(packageName(f), f.filename()));
      } else {
        testFiles.add(f);
      }
    });
```